wire-library/wire-schema/src/commonMain/kotlin/com/squareup/wire/schema/SyntaxRules.kt

```diff
@@ -43,6 +55,20 @@ interface SyntaxRules {
         type: ProtoType,
         label: Field.Label?
       ): Boolean = false
+
+      override fun isTypeOptionalByDefault(
+        protoType: ProtoType,
+        type: Type?,
+        isExtension: Boolean,
+        isOneOf: Boolean
+      ): Boolean = false
+
+      override fun isTypeRequiredByDefault(
+        protoType: ProtoType,
+        type: Type?,
+        isExtension: Boolean,
+        isOneOf: Boolean
+      ): Boolean = false
```

Comment:
Had to break into two methods because for proto2, even if a label is empty, we basically don't wanna change its requirement/optionality.
`isOptional` doesn't mean `!isRequired`, vice versa.